pallets/task/src/lib.rs

```diff
@@ -305,21 +305,31 @@ pub mod pallet {
 		}
 
 		/// Function call that updates a created task.  [ origin, specification, budget, deadline]
-		/// Note: budget specified in this call is added to budget provided in create_task() call.
 		#[pallet::weight(<T as Config>::WeightInfo::update_task(0,0))]
-		pub fn update_task(origin: OriginFor<T>, task_id: T::Hash, title: BoundedVec<u8, T::MaxTitleLen>, specification: BoundedVec<u8, T::MaxSpecificationLen>, 
+		pub fn update_task(origin: OriginFor<T>, task_id: T::Hash, title: BoundedVec<u8, T::MaxTitleLen>, specification: BoundedVec<u8, T::MaxSpecificationLen>,
 			budget: BalanceOf<T>, deadline: u64, attachments: BoundedVec<u8, T::MaxAttachmentsLen>, keywords: BoundedVec<u8, T::MaxKeywordsLen>) -> DispatchResultWithPostInfo {
 
 			// Check that the extrinsic was signed and get the signer.
 			let signer = ensure_signed(origin)?;
 
+			let task = Self::tasks(&task_id).ok_or(<Error<T>>::TaskNotExist)?;
+
 			// Update storage.
 			let _task_id = Self::update_created_task(&signer, &task_id, title, specification, &budget, deadline, attachments, keywords)?;
 
-			// Update balance of escrow account
-			let sub_account = Self::account_id(&task_id);
-			<T as self::Config>::Currency::transfer(&signer, &sub_account, budget,
-				ExistenceRequirement::KeepAlive)?;
+			if task.budget != budget {
+				// Update balance of escrow account
+				let sub_account = Self::account_id(&task_id);
+				if task.budget > budget {
+					let difference = task.budget - budget;
```

Comment:
What happens if the difference is a negative number?

Reply:
the if check above isn't it sufficient to prevent difference to be a negative number?